.github/workflows/gradle-build.yml

```
@@ -0,0 +1,61 @@
+name: Build project
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  build-gradle-project:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: read
+      pull-requests: write
+      contents: read
+      checks: write
+
+    steps:
+      - name: Checkout project sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: 17
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          cache-read-only: ${{ github.ref != 'refs/heads/main' }}
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
+
+      - name: Run build
+        run: ./gradlew build --scan -x deltaCoverage
+
+      - name: Perform Delta coverage
+        if: ${{ github.event_name == 'pull_request' && !cancelled() }}
+        continue-on-error: ${{ contains(github.event.pull_request.labels.*.name, 'suppress-delta-coverage') }}
+        run: ./gradlew deltaCoverage -PdiffBase="refs/remotes/origin/$GITHUB_BASE_REF"
+
+      - name: Append Delta Coverage
+        if: ${{ github.event_name == 'pull_request' && !cancelled() }}
+        continue-on-error: true
+        uses: gw-kit/delta-coverage-plugin/actions/delta-coverage-report@main
+        with:
+          delta-coverage-report: build/reports/coverage-reports/delta-coverage/report.md
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v2.15.1
+        if: ${{ github.event_name == 'pull_request' && !cancelled() }}
```

```suggestion
        if: ${{ !cancelled() }}
```
It's the good idea to publish the test results always